File: backend/doc_configs/design_partner.py
from typing import Optional
from pydantic import BaseModel
from doc_configs.shared import PartialPartyInfo


SYSTEM_PROMPT = """You are a legal document assistant helping a user fill in a Design Partner Agreement Cover Page.

## Fields you must collect

1. **Effective Date** -- when the agreement takes effect
   Ask: "When should this agreement take effect? (e.g. 'today', 'January 1 2027', 'next Monday')"
   For shortcuts like "today", "tomorrow", "Mon", "next Mon", use this lookup table:
{date_lookup}
   For full dates (e.g. "March 15 2026", "2026-03-15"), convert to YYYY-MM-DD format directly.

2. **Term** -- duration of the design partnership
   Ask: "How long should this design partnership last? (e.g. '6 months from Effective Date', '1 year', '90 days')"

3. **Program** -- description of the design partner program activities
   Ask: "What will the design partner program involve? (e.g. 'Beta testing of new analytics dashboard with weekly feedback sessions', 'Early access to API v2 with monthly design reviews')"

4. **Fees** -- fees for the program, if any
   Ask: "Are there any fees for the design partner program? (e.g. 'None', '$500/month', 'Free access during pilot')"

5. **Governing Law** -- jurisdiction whose laws govern the agreement
   Ask: "Which state or jurisdiction's law should govern this agreement? (e.g. 'California', 'New York', 'Delaware')"

6. **Chosen Courts** -- which courts handle disputes
   Ask: "Which courts should have jurisdiction over disputes? (e.g. 'San Francisco, California', 'New York County, New York')"

7. **Party 1 (Provider)** -- name, title, company, notice address, signing date
   Ask for all sub-fields. Example: "Jane Smith, CEO of TechCo, jane@techco.com, signing today"

8. **Party 2 (Partner)** -- same sub-fields as Party 1

## Current document state
{current_fields}

## Instructions
- Work through the fields in order, but adapt naturally to what the user already told you
- Ask 1-2 questions per turn maximum
- Always include a concrete example in parentheses when asking a question
- When the user answers, confirm what you captured then always end with a follow-up question
- Once all fields are collected, tell the user the document is complete and they can download the PDF

## Clearing fields
If the user asks to clear fields (e.g. "clear all", "reset", "clear the date"):
- To clear a specific field, return that field set to its empty value and confirm what was cleared
- To clear all fields, return every field reset to empty values, confirm, then ask the first question again
- Never clear silently

Return JSON with:
- "message": your conversational reply (include examples when asking questions)
- "fields": only the fields you are confident about from this turn; omit everything else
"""


class PartialFields(BaseModel):
    effectiveDate: Optional[str] = None
    term: Optional[str] = None
    program: Optional[str] = None
    fees: Optional[str] = None
    governingLaw: Optional[str] = None
    chosenCourts: Optional[str] = None
    party1: Optional[PartialPartyInfo] = None
    party2: Optional[PartialPartyInfo] = None


class AiResponse(BaseModel):
    message: str
    fields: PartialFields

File: frontend/lib/schemas/design-partner.ts
import { DocSchema } from "../doc-schema";

export const designPartnerSchema: DocSchema = {
  id: "design-partner",
  title: "Design Partner Agreement",
  subtitle: "Cover Page",
  infoBox:
    'This Design Partner Agreement consists of: (1) this Cover Page and (2) the Common Paper Design Partner Agreement Standard Terms Version 1.0 identical to those posted at commonpaper.com/standards/design-partner-agreement/1.0. Any modifications of the Standard Terms should be made on the Cover Page, which will control over conflicts with the Standard Terms.',
  coverFields: [
    {
      key: "effectiveDate",
      label: "Effective Date",
      type: "date",
      placeholder: "Select a date",
    },
    {
      key: "term",
      label: "Term",
      hint: "Duration of the design partnership",
      type: "text",
      placeholder: "e.g. 6 months from Effective Date",
    },
    {
      key: "program",
      label: "Program",
      hint: "Description of the design partner program activities",
      type: "textarea",
      placeholder: "Describe the program activities...",
    },
    {
      key: "fees",
      label: "Fees",
      hint: "Fees for the design partner program, if any",
      type: "text",
      placeholder: "e.g. None, or $500/month",
    },
    {
      key: "governingLaw",
      label: "Governing Law",
      type: "text",
      placeholder: "State",
    },
    {
      key: "chosenCourts",
      label: "Chosen Courts",
      type: "text",
      placeholder: "City/county and state",
    },
  ],
  parties: [
    { label: "Provider", fields: ["name", "title", "company", "noticeAddress", "date"] },
    { label: "Partner", fields: ["name", "title", "company", "noticeAddress", "date"] },
  ],
  signingNote: "By signing this Cover Page, each party agrees to enter into this Design Partner Agreement as of the Effective Date.",
  templateFile: "templates/design-partner-agreement.md",
  footerNote: "Common Paper Design Partner Agreement (Version 1.0) free to use under CC BY 4.0.",
  greeting:
    "Hi! I'm your Design Partner Agreement assistant. I'll help you set up a design partner arrangement for early-stage product development. This covers product access, feedback, confidentiality, and IP ownership. Let's start -- when should this agreement take effect? (e.g. 'Today', 'January 1 2027', 'Next Monday')",
  pdfFilename: "design-partner-agreement.pdf",
};
